File: Trees/TreeTraversal.js
class Node {
	constructor(value) {
		this.left = null;
		this.data = value;
		this.right = null;
	}
}

function preOrder(root) {
    if(root !== null){
        console.log(root.data);
        preOrder(root.left)
        preOrder(root.right)
    }
}

root = new Node(1);
root.left = new Node(3);
root.right = new Node(5);
root.left.left = new Node(2);
root.left.right = new Node(4);
root.right.right = new Node(8);

preOrder(root)

File: Trees/TreeTraversal.py
class Node:
    def __init__(self, value):
        self.left = None
        self.data = value
        self.right = None
        
        
def preOrderTraversal(root):
    if(root != None):
        print(root.data, end=" -> ")
        preOrderTraversal(root.left)
        preOrderTraversal(root.right)
    
def inOrderTraversal(root):
    if(root != None):
        inOrderTraversal(root.left)
        print(root.data, end=" -> ")
        inOrderTraversal(root.right)
    
def postOrderTraversal(root):
    if(root != None):
        postOrderTraversal(root.left)
        postOrderTraversal(root.right)
        print(root.data, end=" -> ")
    
root = Node(1)
root.left = Node(3)
root.right = Node(5)
root.left.left = Node(2)
root.left.right = Node(4)
root.right.right = Node(8)

print("Pre Order")
preOrderTraversal(root)
print()
print("In Order")
inOrderTraversal(root)
print()
print("Post Order")
postOrderTraversal(root)